### PR TITLE
Move docker resource limit settings from server to agent

### PR DESCRIPTION
**Pull request by 6543**
so you can set it per agent and not per server

related to  #3156 to cleanup the step definition

### CODE REVIEW

File: pipeline/backend/docker/flags.go [hunk 1]
```diff
@@ -56,4 +56,37 @@ var Flags = []cli.Flag{
...
+	},
+	&cli.Int64Flag{
+		EnvVars: []string{"WOODPECKER_BACKEND_DOCKER_LIMIT_MEM", "WOODPECKER_LIMIT_MEM"},
+		Name:    "backend-docker-limit-mem",
+		Usage:   "maximum memory allowed in bytes",
+	},
+	&cli.Int64Flag{
+		EnvVars: []string{"WOODPECKER_BACKEND_DOCKER_LIMIT_SHM_SIZE", "WOODPECKER_LIMIT_SHM_SIZE"},
+		Name:    "backend-docker-limit-shm-size",
+		Usage:   "docker compose /dev/shm allowed in bytes",
```
Comment (6543) [inline pipeline/backend/docker/flags.go:75]:
```suggestion
		Usage:   "docker /dev/shm allowed in bytes",
```

**Comment by lafriks** [comment]
Maybe server can still provide defaults?

**Comment by 6543** [comment]
no I'm against this ...

we e.g. set the tmp path for the local backend on the server side

**Comment by lafriks** [comment]
I have no strong opinion about this but that would avoid it being breaking change

**Comment by anbraten** [comment]
I think it the default fits well into the agent settings, specific settings however should probably be still settable via the pipeline config. We just need to move it into the backend config section.

**Comment by 6543** [comment]
well make it non breaking for now then ... but add a TODO comment so we can do so on next majour bump

also I add a docker specific config like with kubernetis for limiting per step

File: pipeline/backend/docker/flags.go [hunk 2]
```diff
@@ -56,4 +56,37 @@ var Flags = []cli.Flag{
 		Name:    "backend-docker-volumes",
 		Usage:   "backend docker volumes (comma separated)",
 	},
+	//
+	// resource limit parameters
+	//
+	&cli.IntFlag{
+		Sources: cli.EnvVars("WOODPECKER_BACKEND_DOCKER_LIMIT_MEM_SWAP", "WOODPECKER_LIMIT_MEM_SWAP"),
+		Name:    "backend-docker-limit-mem-swap",
+		Usage:   "maximum swappable memory allowed in bytes",
```
Comment (6543) [inline pipeline/backend/docker/flags.go:65]:
```suggestion
		Usage:   "maximum memory used for swap in bytes",
```

File: docs/docs/91-migrations.md [hunk 3]
```diff
@@ -4,6 +4,7 @@ Some versions need some changes to the server configuration or the pipeline conf
 
 ## `next`
 
+- Move docker resource limit settings into agent configuration
```
Comment (6543) [inline docs/docs/91-migrations.md:7]:
```suggestion
- Move docker resource limit settings from server into agent configuration
```

**Review by pat-s — APPROVED** [review]
Fits better into the agent than server but I agree, it should be settable in the step config like for k8s, as every container usually has a different requirement.

Given it's breaking due to the move, it fits well into 3.0.

**Comment by 6543** [comment]
yes have it alser per stes should be an backend option like done for kubes settings. this will be a new issue to address ...

**Pull request merged**